lib/Compile.js

```diff
@@ -94,13 +94,7 @@ function compileSources(
       if (exitCode === 0) {
         resolve();
       } else {
-        const msg =
-          'Compilation failed while attempting to ' +
-          (testFilePaths.length > 0
-            ? 'build ' + testFilePaths.join(' ')
-            : 'run `elm make` on ' + projectRootDir);
-
-        reject(msg);
+        reject(new Error(`\`elm make\` failed with exit code ${exitCode}.`));
```

Comment:
The old error was super verbose and not useful when the nice error from `elm make` is printed just above anyway.